src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgAddressTypeChangedCeEstProcessor.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.ons.census.fwmt.outcomeservice.converter.spg;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+import uk.gov.ons.census.fwmt.outcomeservice.config.GatewayOutcomeQueueConfig;
+import uk.gov.ons.census.fwmt.outcomeservice.converter.SpgOutcomeServiceProcessor;
+import uk.gov.ons.census.fwmt.outcomeservice.data.GatewayCache;
+import uk.gov.ons.census.fwmt.outcomeservice.dto.SpgOutcomeSuperSetDto;
+import uk.gov.ons.census.fwmt.outcomeservice.message.GatewayOutcomeProducer;
+import uk.gov.ons.census.fwmt.outcomeservice.service.impl.GatewayCacheService;
+import uk.gov.ons.census.fwmt.outcomeservice.template.TemplateCreator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.CESPG_ADDRESS_TYPE_CHANGED_OUTCOME_SENT;
+import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.CESPG_OUTCOME_SENT;
+import static uk.gov.ons.census.fwmt.outcomeservice.enums.EventType.ADDRESS_TYPE_CHANGED_CEEST;
+import static uk.gov.ons.census.fwmt.outcomeservice.enums.SurveyType.spg;
+
+@Component("ADDRESS_TYPE_CHANGED_CE_EST")
+public class SpgAddressTypeChangedCeEstProcessor implements SpgOutcomeServiceProcessor {
+
+  @Autowired
+  private GatewayOutcomeProducer gatewayOutcomeProducer;
+
+  @Autowired
+  private GatewayEventManager gatewayEventManager;
+
+  @Autowired
+  private GatewayCacheService gatewayCacheService;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    cacheData(outcome, caseId);
+
+    Map<String, Object> root = new HashMap<>();
+    String eventDateTime = outcome.getEventDate().toString();
```

Comment:
Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

Reply:
Done.